src/gui/vector/qgsattributesformmodel.h

```diff
@@ -0,0 +1,738 @@
+/***************************************************************************
+    qgsattributesformmodel.h
+    ---------------------
+    begin                : March 2025
+    copyright            : (C) 2025 by Germán Carrillo
+    email                : german at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSATTRIBUTESFORMMODEL_H
+#define QGSATTRIBUTESFORMMODEL_H
+
+// We don't want to expose this in the public API
+#define SIP_NO_FILE
+
+#include "qgsaddtaborgroup.h"
+#include "qgsattributeeditorelement.h"
+#include "qgsoptionalexpression.h"
+#include "qgsvectorlayer.h"
+
+#include <QAbstractItemModel>
+#include <QPushButton>
+
+/**
+ * \brief Class to describe editor data contained within a QgsAttributesFormModel.
+ *
+ * \warning Not part of stable API and may change in future QGIS releases.
+ * \ingroup gui
+ * \note Prior to QGIS 3.44 this was available as QgsAttributesFormProperties
+ * \since QGIS 3.44
+ */
+class GUI_EXPORT QgsAttributesFormTreeData
```

Comment:
Let's remove the tree reference.

```suggestion
class GUI_EXPORT QgsAttributesFormData
```